Gemfile

```diff
@@ -4,5 +4,8 @@ group :test do
   gem 'rake'
   gem 'rspec'
   gem 'beaker', '~>1.17.0'
+  if ENV['GEM_SOURCE'] =~ /rubygems\.delivery\.puppetlabs\.net/
```

Comment:
I like that better :)  Not sure the regex is necessary, and if I was going to use one, I'd make it more strict than this (using a leading ^ and trailing $), but it addresses the main concern that I had.